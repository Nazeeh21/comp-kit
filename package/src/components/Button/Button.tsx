import { styled } from '@stitches/react';

export const Button = styled('button', {
  backgroundColor: 'white',
  borderRadius: '6px',
  fontSize: '14px',
  padding: '10px 15px',
  cursor: 'pointer',
  border: 'none',
  '&:hover': {
    backgroundColor: '#f5f5f5',
  },

  variants: {
    color: {
      dark: {
        backgroundColor: 'black',
        color: 'white',
        '&:hover': {
          backgroundColor: '#1b1a1b',
        },
        '&:focus': {
          outline: '#1b1a1b solid 2px',
          outlineOffset: '1px',
        },
      },
      light: {
        backgroundColor: 'white',
        color: 'black',
        '&:hover': {
          backgroundColor: '#f5f5f5',
        },
        '&:focus': {
          outline: '#f5f5f5 solid 2px',
          outlineOffset: '1px',
        },
      },
    },
  },
  defaultVariants: {
    color: 'light',
  },
});
