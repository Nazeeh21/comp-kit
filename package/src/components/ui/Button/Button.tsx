import { styled } from '@stitches/react';

const _DarkBackgroundColor = '#222122';
const _LightBackgroundColor = '#f9fdfe';

export const Button = styled('button', {
  backgroundColor: 'white',
  borderRadius: '6px',
  fontSize: '14px',
  padding: '10px 15px',
  width: 'fit-content',
  cursor: 'pointer',
  border: 'none',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  '&:hover': {
    backgroundColor: '#f5f5f5',
  },

  variants: {
    color: {
      dark: {
        backgroundColor: _DarkBackgroundColor,
        color: 'white',
        '&:hover': {
          backgroundColor: '#1b1a1b',
        },
        '&:focus': {
          outline: '#1b1a1b solid 2px',
          outlineOffset: '1px',
        },
      },
      light: {
        backgroundColor: _LightBackgroundColor,
        color: 'black',
        '&:hover': {
          backgroundColor: '#f5f5f5',
        },
        '&:focus': {
          outline: '#f5f5f5 solid 2px',
          outlineOffset: '1px',
        },
      },
    },
  },
  defaultVariants: {
    color: 'light',
  },
});
