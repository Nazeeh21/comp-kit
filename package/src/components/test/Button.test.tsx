import { Button } from '../Button/Button';
import React from 'react';
import renderer from 'react-test-renderer';

describe('index', () => {
  it('renders button', () => {
    const container = renderer.create(<Button>Test Button</Button>).toJSON();
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
