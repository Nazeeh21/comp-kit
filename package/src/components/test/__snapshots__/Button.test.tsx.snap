// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders button 1`] = `
<button
  className="c-kexgoD c-kexgoD-dRTyXt-color-light"
>
  Test Button
</button>
`;
